test/e2e/installplan_e2e_test.go

```diff
@@ -642,6 +642,203 @@ func TestCreateInstallPlanWithPreExistingCRDOwners(t *testing.T) {
+			defer cleanupCatalogSource()
+
+			// Attempt to get the catalog source before creating install plan(s)
+			_, err := fetchCatalogSource(t, crc, mainCatalogSourceName, testNamespace, catalogSourceRegistryPodSynced)
+			require.NoError(t, err)
+
+			subscriptionName := genName("sub-nginx-")
+			// this subscription will be cleaned up below without the clean up function
+			createSubscriptionForCatalog(t, crc, testNamespace, subscriptionName, mainCatalogSourceName, mainPackageName, stableChannel, "", v1alpha1.ApprovalAutomatic)
+
+			subscription, err := fetchSubscription(t, crc, testNamespace, subscriptionName, subscriptionHasInstallPlanChecker)
+			require.NoError(t, err)
+			require.NotNil(t, subscription)
+
+			installPlanName := subscription.Status.Install.Name
+
+			// Wait for InstallPlan to be status: Complete or failed before checking resource presence
+			completeOrFailedFunc := buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete, v1alpha1.InstallPlanPhaseFailed)
+			fetchedInstallPlan, err := fetchInstallPlan(t, crc, installPlanName, completeOrFailedFunc)
+			require.NoError(t, err)
+			t.Logf("Install plan %s fetched with status %s", fetchedInstallPlan.GetName(), fetchedInstallPlan.Status.Phase)
+			require.Equal(t, v1alpha1.InstallPlanPhaseComplete, fetchedInstallPlan.Status.Phase)
+
+			// Ensure that the desired resources have been created
+			expectedSteps := map[registry.ResourceKey]struct{}{
+				registry.ResourceKey{Name: mainCRD.Name, Kind: "CustomResourceDefinition"}:              {},
+				registry.ResourceKey{Name: mainPackageStable, Kind: v1alpha1.ClusterServiceVersionKind}: {},
+			}
+
+			require.Equal(t, len(expectedSteps), len(fetchedInstallPlan.Status.Plan), "number of expected steps does not match installed")
+
+			for _, step := range fetchedInstallPlan.Status.Plan {
+				key := registry.ResourceKey{
+					Name: step.Resource.Name,
+					Kind: step.Resource.Kind,
+				}
+				_, ok := expectedSteps[key]
+				require.True(t, ok, "couldn't find %v in expected steps: %#v", key, expectedSteps)
+
+				// Remove the entry from the expected steps set (to ensure no duplicates in resolved plan)
+				delete(expectedSteps, key)
+			}
+
+			// Should have removed every matching step
+			require.Equal(t, 0, len(expectedSteps), "Actual resource steps do not match expected")
+
+			// new CRD changes
+			if tt.multipleVersions != nil {
+				for _, version := range tt.multipleVersions {
+					crdVersion := apiextensions.CustomResourceDefinitionVersion{
+						Name:    version,
+						Served:  true,
+						Storage: false,
+					}
+					mainCRD.Spec.Versions = append(mainCRD.Spec.Versions, crdVersion)
+				}
+				mainCRD.Spec.Version = ""
+				mainCRD.Spec.Versions[len(tt.multipleVersions)-1].Storage = true
+			} else {
+				mainCRD.Spec.Version = tt.version
+			}
+
+			var min float64 = 2
+			var max float64 = 256
+			if tt.changeSchema == true {
```

Comment:
also non-blocking - this may just be a personal thing, but I find it easier to reason about table-driven tests when there are minimal conditionals in the test. 

e.g. instead of 

```go
test := {
   branch: true
}

t.Run() {
  if branch {
    do(thing)
  }
  do(other thing)
}
```

doing something like:

```go 
func branch(thing) {
  return other thing
}

tests := {
   {
   obj: branch(thing)
  },
  {
   obj: thing
  }
}

t.Run() {
  do(tt.obj)
}
```

Reply:
No, you're right. I've fixed it now.